Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -12,18 +12,87 @@ static void Main(string[] args)
             Test("42*?47=1974", -1);
             Test("2*12?=247", -1);
             Console.ReadKey(true);
+            
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
+        public static int FindDigit(string equation)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            int l = equation.Length;
+            int m = l+1;
+            int x = 0;
+            char[] ch = equation.ToCharArray();
+            
+             for(int i = 0 ; i < l ; i++){
+                if(ch[i] == '?' && i >= (l/2 + 1)){
+                    string[] str = equation.Split("=");
+                    string s = str[1];
+                    string str1 = str[0];
+                    string[] str2 = str1.Split("*");
+                    string str3 = str2[0];
+                    string str4 = str2[1];
+                    int num1 = Int32.Parse(str3);
+                    int num2 = Int32.Parse(str4);
+                    int result = num1 * num2;
+                    string result1 = result.ToString();
+                    char[] ch1 = s.ToCharArray();
+                    char[] ch2 = result1.ToCharArray();
+                    char res = ' ';
+                    for(int j = 0 ; j < s.Length ; j++){
+                        if(ch1[j] == '?'){
+                            res = ch2[j];
+                        }
+                    }
+                    x = res - '0';
+                    return x;
+                }else if(ch[i] == '?' && i < (l/2 + 1)){
+                    string[] str = equation.Split("=");

```

Comment:
code is very clumsy and not readable. Please use meaningful variable names and try to create small private methods so that code is neat and clean.